packages/compiler/src/compiler/base/nodes/tag.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  isContentTag,
+  isMessageTag,
+  isRefTag,
+  isToolCallTag,
+} from '$compiler/compiler/utils'
+import errors from '$compiler/error/errors'
+import {
+  ContentTag,
+  ElementTag,
+  MessageTag,
+  ReferenceTag,
+  ToolCallTag,
+} from '$compiler/parser/interfaces'
+
+import { CompileNodeContext } from '../types'
+import { compile as resolveContent } from './tags/content'
+import { compile as resolveMessage } from './tags/message'
+import { compile as resolveRef } from './tags/ref'
+import { compile as resolveToolCall } from './tags/toolCall'
```

Comment:
Not to change but it would be better if from the file you export these names. No need for doing this and also when you are in the file you understand what type of compile is

Reply:
Hmm I like that all nodes just need to define a generic `compile` method, same name and arguments.

Reply:
Ok if you like it no more discussion xD